File: Linked List/linkedlist.cpp
#include <iostream>
using namespace std;

struct node{
    int data;
    node* next;
};

class linked_list{
    private:
        node* head;
        node* tail;
    public:

        linked_list(){
            head = NULL;
            tail = NULL;
        }

        void push(int val){

            node* current = new node;

            current->data = val;
            current->next = NULL;


            if ( head == NULL && tail == NULL ){
                head = current; tail = current;
            }
            else{
                tail->next = current;
                tail = current;
            }
            
        }

        void display(){
            
            node* current = head;

            while ( current != NULL ){

                cout<<current->data<<endl;
                current = current->next;

            }
            cout<<"------------------"<<endl;

        }

        void pop(){

            node* temp = tail;
            node* current = head; 
            while ( current->next != NULL )
            {
                tail = current;
                current = current->next;
            }
            
            tail->next = NULL;
            free(temp);

        }
};

int main(){

    linked_list l1;

    l1.push(4);
    l1.push(2);
    l1.push(7);
    

    l1.display();
    l1.pop();
    l1.display();
    
    return 0;
}

File: Linked List/linkedlist.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:

    def __init__(self, head=None, tail=None):
        self.head = head
        self.tail = tail

    def push(self, data):

        if self.head == None:
            self.head = Node(data)
            self.tail = self.head

        else:
            
            new_node = Node(data)
            new_node.next = None
            self.tail.next = new_node #type: ignore
            self.tail = new_node

    def display(self):
            
        temp = self.head

        while temp != None:
            print(temp.data)
            temp = temp.next

    def pop(self):

        temp = self.head

        while temp.next != self.tail:
            temp = temp.next

        temp.next = None
        self.tail = temp


if __name__ == "__main__":
        
        llist = LinkedList()
        llist.push(1)
        llist.push(2)
        llist.push(3)
        llist.push(4)
        llist.pop()
        llist.push(5)
    
        llist.display()